File: grpc-server/Productores.py
import json
from kafka import KafkaProducer  


    ### Productor kafka popularidadReceta
def popularidadReceta(idReceta, signo):
    producer = KafkaProducer(bootstrap_servers='localhost:9092')  
    mkafka = {
        "idReceta": idReceta, 
        "popularidad" : signo
    }
    #signo + para agregar 1 o - para restar 1
    print("este es el json") 
    print(mkafka)
    producer.send(topic = 'popularidadReceta', value=json.dumps(mkafka).encode("utf-8"))
    producer.close()

### Productor kafka popularidadUsuario
def popularidadUsuario(usuario, signo):
    producer = KafkaProducer(bootstrap_servers='localhost:9092')  
    mkafka = {
        "usuario": usuario, 
        "popularidad" : signo
    }
    #signo + para agregar 1 o - para restar 1
    print("este es el json") 
    print(mkafka)
    producer.send(topic = 'popularidadUsuario', value=json.dumps(mkafka).encode("utf-8"))
    producer.close()

### Productor kafka comentario
def comentario(usuario, idReceta, comentario):
    producer = KafkaProducer(bootstrap_servers='localhost:9092')  
    mkafka = {
        "usuario": usuario, 
        "idReceta" : idReceta,
        "comentario" : comentario
    }
    #signo + para agregar 1 o - para restar 1
    print("este es el json") 
    print(mkafka)
    producer.send(topic = 'comentario', value=json.dumps(mkafka).encode("utf-8"))
    producer.close()


if __name__ == '__main__':
    pass
    for i in range(40):
        popularidadReceta(12, "+")
        popularidadReceta(13, "+")

File: agregar/Productores.js

// Productor kafka popularidadReceta

const { Kafka } = require('kafkajs');

async function popularidadReceta(idReceta, signo) {
  const kafka = new Kafka({
    clientId: 'my-app',
    brokers: ['localhost:9092']
  });

  const producer = kafka.producer();

  const mkafka = {
    idReceta: idReceta,
    popularidad: signo
  };

  console.log('Este es el JSON');
  console.log(mkafka);

  await producer.connect();
  await producer.send({
    topic: 'popularidadReceta',
    messages: [{ value: JSON.stringify(mkafka) }]
  });

  await producer.disconnect();
}

// Ejemplo de uso:
popularidadReceta(123, '+1');



//Productor kafka popularidadUsuario

const { Kafka } = require('kafkajs');

async function popularidadUsuario(usuario, signo) {
  const kafka = new Kafka({
    clientId: 'my-app',
    brokers: ['localhost:9092']
  });

  const producer = kafka.producer();

  const mkafka = {
    usuario: usuario,
    popularidad: signo
  };

  console.log('Este es el JSON');
  console.log(mkafka);

  await producer.connect();
  await producer.send({
    topic: 'popularidadUsuario',
    messages: [{ value: JSON.stringify(mkafka) }]
  });

  await producer.disconnect();
}

// Ejemplo de uso:
popularidadUsuario('juan', '+');



// Productor kafka comentario

const { Kafka } = require('kafkajs');

async function comentario(usuario, idReceta, comentario) {
  const kafka = new Kafka({
    clientId: 'my-app',
    brokers: ['localhost:9092']
  });

  const producer = kafka.producer();

  const mkafka = {
    usuario: usuario,
    idReceta: idReceta,
    comentario: comentario
  };

  console.log('Este es el JSON');
  console.log(mkafka);

  await producer.connect();
  await producer.send({
    topic: 'comentario',
    messages: [{ value: JSON.stringify(mkafka) }]
  });

  await producer.disconnect();
}

// Ejemplo de uso:
comentario('juan', 123, 'Este es un comentario de ejemplo');
